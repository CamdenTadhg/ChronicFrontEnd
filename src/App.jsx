import React from 'react';
import {BrowserRouter, Routes, Route} from 'react-router-dom';
import ProtectedRoutes from './components/ProtectedRoutes';
import NavBar from './components/NavBar';
import Home from './components/Home';
import Latest from './components/Latest';
import Profile from './components/Profile';
import About from './components/About';
import Tracking from './components/Tracking';
import Data from './components/Data';


function App() {

  return (
    <>
      <BrowserRouter>
        <NavBar className='fixed-top'/>
        <Routes>
          <Route path="/latest" element={<Latest/>}/>
          <Route path="/about" element={<About/>}/>
          <Route element={<ProtectedRoutes/>}>
            <Route path='/profile' element={<Profile/>}/>
            <Route path='/tracking' element={<Tracking/>}/>
            <Route path='/data' element={<Data/>}/>
          </Route>
          <Route path="/" element={<Home/>} />
        </Routes>
      </BrowserRouter>
    </>
  )
}

export default App
