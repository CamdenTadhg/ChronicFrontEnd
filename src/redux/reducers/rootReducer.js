import {combineReducers} from 'redux';
import profileReducer from './profileReducer';
import trackingReducer from './trackingReducer';
import dataReducer from './dataReducer';
import latestReducer from './latestReducer';

const rootReducer = combineReducers({
    profile: profileReducer,
    tracking: trackingReducer,
    data: dataReducer,
    latest: latestReducer
});

export default rootReducer;