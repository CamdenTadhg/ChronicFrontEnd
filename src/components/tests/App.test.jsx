import React, { act } from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react';
import {it, expect, vi} from 'vitest';
import '@testing-library/jest-dom';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import {MemoryRouter, Route, Routes, Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import App from '../../App';
import rootReducer from '../../redux/reducers/rootReducer';
import ChronicAPI from '../../api/chronicAPI';


it('renders without errors', async () => {
    const store = configureStore({reducer: rootReducer});
    ChronicAPI.getArticleIds = vi.fn();
    ChronicAPI.getArticleIds.mockResolvedValue([1234,2233]);
    ChronicAPI.getArticles = vi.fn();
    ChronicAPI.getArticles.mockResolvedValue([
        {PMID: 2345, title: 'a test article', abstract: "this is a test article"},
        {PMID: 5732, title: 'a second test article', abstract: "this is also a test article"}
    ]);
    await waitFor(() => {
        render(
            <Provider store={store}>
                <App/>
            </Provider>
        )
    });
});

it('matches the snapshot', async () => {
    const store = configureStore({reducer: rootReducer});
    ChronicAPI.getArticleIds = vi.fn();
    ChronicAPI.getArticleIds.mockResolvedValue([1234,2233]);
    ChronicAPI.getArticles = vi.fn();
    ChronicAPI.getArticles.mockResolvedValue([
        {PMID: 2345, title: 'a test article', abstract: "this is a test article"},
        {PMID: 5732, title: 'a second test article', abstract: "this is also a test article"}
    ]);
    const app = render(
        <Provider store={store}>
            <App/>
        </Provider>
    );
    await waitFor(() => expect(app).toMatchSnapshot());
});

it('allows a logged in user to navigate to protected routes', async () => {
    const store = configureStore({reducer: rootReducer, preloadedState: {
        profile: {
            userId: 7,
            diagnoses: [ {
                diagnosis: 'chronic fatigue syndrome',
                keywords: ['ME']
            }]
        }
    }});
    ChronicAPI.getArticleIds = vi.fn();
    ChronicAPI.getArticleIds.mockResolvedValue([1234,2233]);
    ChronicAPI.getArticles = vi.fn();
    ChronicAPI.getArticles.mockResolvedValue([
        {PMID: 2345, title: 'a test ME article', abstract: 'this is a test article about ME'},
        {PMID: 5732, title: 'a second test ME article', abstract: 'this is also a test article about ME'}
    ]);
    const {getByText} = render(
        <Provider store={store}>
            <App/>
        </Provider>
    );
    fireEvent.click(getByText('Tracking'));
    await waitFor(() => expect(getByText('Tracking page coming soon...')).toBeInTheDocument());
    ;
});

it('redirects an anonymous user from accessing protected routes', async () => {
    const store = configureStore({reducer: rootReducer});
    ChronicAPI.getArticleIds = vi.fn();
    ChronicAPI.getArticleIds.mockResolvedValue([1234,2233]);
    ChronicAPI.getArticles = vi.fn();
    ChronicAPI.getArticles.mockResolvedValue([
        {PMID: 2345, title: 'a test article', abstract: 'this is a test article'},
        {PMID: 5732, title: 'a second test article', abstract: 'this is also a test article'}
    ]);
    const {getByText} = render(
        <Provider store={store}>
            <App/>
        </Provider>
    );
    act(() => {
        window.history.pushState({}, 'Tracking', '/tracking');
    });
    await waitFor(() => expect(getByText('Your Health. Your Data. Your Power.')).toBeInTheDocument());
});