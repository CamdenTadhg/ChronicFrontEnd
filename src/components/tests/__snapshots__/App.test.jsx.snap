// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <nav
          class="navbar navbar-expand-lg bg-secondary fixed-top"
          full="true"
        >
          <div
            class="container-fluid"
          >
            <a
              class="brand navbar-brand"
            >
              <img
                alt="Logo"
                class="chronic-logo"
                src="/src/assets/Chronic.png"
              />
              <h6
                class="logo-text"
              >
                CHRONIC
              </h6>
            </a>
            <ul
              class="nav"
            >
              <a
                class="nav-link"
                href="/about"
              >
                About
              </a>
              <button
                class="nav-link"
              >
                Login
              </button>
              <button
                class="nav-link"
              >
                Signup
              </button>
            </ul>
          </div>
        </nav>
      </nav>
      <div
        class="p-0 container-fluid"
      >
        <div
          class="navbar-space m-0 p-0 row"
        />
        <div
          class="main-body m-0 p-0 row"
        >
          <div
            class="home col-9"
          >
            <div
              class="out-home-page"
            >
              <div
                class="hero-text"
              >
                Your Health. Your Data. Your Power.
              </div>
              <img
                alt="Logo"
                class="hero-image"
                src="/src/assets/Chronic_dark.png"
              />
              <div
                class="hero-text"
              >
                CHRONIC
              </div>
            </div>
          </div>
          <div
            class="sidebar col-3"
          >
            <a
              href="/latest"
            >
              <h2
                class="latest-head"
              >
                The Latest
              </h2>
            </a>
            <div
              class="articles-sidebar"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <nav
        class="navbar navbar-expand-lg bg-secondary fixed-top"
        full="true"
      >
        <div
          class="container-fluid"
        >
          <a
            class="brand navbar-brand"
          >
            <img
              alt="Logo"
              class="chronic-logo"
              src="/src/assets/Chronic.png"
            />
            <h6
              class="logo-text"
            >
              CHRONIC
            </h6>
          </a>
          <ul
            class="nav"
          >
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
            <button
              class="nav-link"
            >
              Login
            </button>
            <button
              class="nav-link"
            >
              Signup
            </button>
          </ul>
        </div>
      </nav>
    </nav>
    <div
      class="p-0 container-fluid"
    >
      <div
        class="navbar-space m-0 p-0 row"
      />
      <div
        class="main-body m-0 p-0 row"
      >
        <div
          class="home col-9"
        >
          <div
            class="out-home-page"
          >
            <div
              class="hero-text"
            >
              Your Health. Your Data. Your Power.
            </div>
            <img
              alt="Logo"
              class="hero-image"
              src="/src/assets/Chronic_dark.png"
            />
            <div
              class="hero-text"
            >
              CHRONIC
            </div>
          </div>
        </div>
        <div
          class="sidebar col-3"
        >
          <a
            href="/latest"
          >
            <h2
              class="latest-head"
            >
              The Latest
            </h2>
          </a>
          <div
            class="articles-sidebar"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <nav
          class="navbar navbar-expand-lg bg-secondary fixed-top"
          full="true"
        >
          <div
            class="container-fluid"
          >
            <a
              class="brand navbar-brand"
            >
              <img
                alt="Logo"
                class="chronic-logo"
                src="/src/assets/Chronic.png"
              />
              <h6
                class="logo-text"
              >
                CHRONIC
              </h6>
            </a>
            <ul
              class="nav"
            >
              <a
                class="nav-link"
                href="/about"
              >
                About
              </a>
              <button
                class="nav-link"
              >
                Login
              </button>
              <button
                class="nav-link signup"
              >
                Signup
              </button>
            </ul>
          </div>
        </nav>
      </nav>
      <div
        class="p-0 container-fluid"
      >
        <div
          class="navbar-space m-0 p-0 row"
        />
        <div
          class="main-body m-0 p-0 row"
        >
          <div
            class="home col-9"
          >
            <div
              class="out-home-page"
            >
              <div
                class="hero-text"
              >
                Your Health. Your Data. Your Power.
              </div>
              <img
                alt="Logo"
                class="hero-image"
                src="/src/assets/Chronic_dark.png"
              />
              <div
                class="hero-text"
              >
                CHRONIC
              </div>
            </div>
          </div>
          <div
            class="sidebar col-3"
          >
            <a
              href="/latest"
            >
              <h2
                class="latest-head"
              >
                The Latest
              </h2>
            </a>
            <div
              class="articles-sidebar"
            >
              <p>
                An error occurred. Please try again later
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <nav
        class="navbar navbar-expand-lg bg-secondary fixed-top"
        full="true"
      >
        <div
          class="container-fluid"
        >
          <a
            class="brand navbar-brand"
          >
            <img
              alt="Logo"
              class="chronic-logo"
              src="/src/assets/Chronic.png"
            />
            <h6
              class="logo-text"
            >
              CHRONIC
            </h6>
          </a>
          <ul
            class="nav"
          >
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
            <button
              class="nav-link"
            >
              Login
            </button>
            <button
              class="nav-link signup"
            >
              Signup
            </button>
          </ul>
        </div>
      </nav>
    </nav>
    <div
      class="p-0 container-fluid"
    >
      <div
        class="navbar-space m-0 p-0 row"
      />
      <div
        class="main-body m-0 p-0 row"
      >
        <div
          class="home col-9"
        >
          <div
            class="out-home-page"
          >
            <div
              class="hero-text"
            >
              Your Health. Your Data. Your Power.
            </div>
            <img
              alt="Logo"
              class="hero-image"
              src="/src/assets/Chronic_dark.png"
            />
            <div
              class="hero-text"
            >
              CHRONIC
            </div>
          </div>
        </div>
        <div
          class="sidebar col-3"
        >
          <a
            href="/latest"
          >
            <h2
              class="latest-head"
            >
              The Latest
            </h2>
          </a>
          <div
            class="articles-sidebar"
          >
            <p>
              An error occurred. Please try again later
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
